src/App.js

```diff
@@ -1,15 +1,104 @@
 import SearchBar from './components/SearchBar/SearchBar';
 import Users from './components/Users/Users';
 import './App.css';
+import { useEffect, useState } from 'react';
+import NoResult from './components/NoResult/NoResult';
+import Loading from './components/Loading/Loading';
+import Error from './components/Error/Error';
+
+const API = "https://users-app-backend.onrender.com/users"
 
 function App() {
+  const [ users, setUsers ] = useState([]);
+  const [ searchInput, setSearchInput ] = useState("");
+  const [ expanded, setExpanded ] = useState([]);
+  const [ loading, setLoading ] = useState(false);
+  const [ error, setError ] = useState("");
+
   // TODO: Fetch data here
+  useEffect(()=> {
+    async function fetchData() {
+      try {
+        setError("");
+        setLoading(true);
+        const response = await fetch(`${API}`);
+        const json = await response.json();
+        const { data, error } = json;
+        if(response.ok) {
+          setUsers(data);
+          setLoading(false);
+        } else {
+          setError(error);
+          setLoading(false);
+        }
+        
+      } catch (error) {
+        setError(error.meassage);
+        setLoading(false);
+      }
+    }
+    fetchData();
+  }, [])
+
+  
+
+  let dataToDisplay = users;
+
+  if(searchInput) {
+    dataToDisplay = users.filter(user => {
+      const { name, country, company } = user;
+      return (name.toLowerCase().includes(searchInput.toLowerCase()) || country.toLowerCase().includes(searchInput.toLowerCase()) || company.toLowerCase().includes(searchInput.toLowerCase()));
+    })
+  }
+
+  const handleToggleExpanded = (id) => {
+    if(!expanded.includes(id)) {
+      const newExpanded = [...expanded, id];
+      setExpanded(newExpanded);
+    } else {
+      const removed = expanded.filter(currId => currId !== id);
+      setExpanded(removed);
+    }
+  }
+
+  const handleExpandAll = () => {
+    const allIds = users.map(user => user.id);
+    setExpanded(allIds);
+  }
+
+  const handleCollapseAll = () => {
+    setExpanded([]);
+  }
+
+  const renderContent = () => {
+    if(loading) {
+      return <Loading />
+    } else if(error) {
+      return <Error error={error} />
+    } else {
+      return <Users 
+      users={dataToDisplay} 
+      searchInput={searchInput} 
```

Comment:
I think this is the only choice that I disagree with. We're doing the filtering already up here in `<App />`. So I don't think that `<Users />` or `<User />` will need this information.

Reply:
Are you just referring to searchInput? If so, I got it now.